src/test/java/ca/team2706/frc/robot/sensors/AnalogSelectorTest.java

```diff
@@ -0,0 +1,66 @@
+package ca.team2706.frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Mocked;
+import mockit.Tested;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Class to test the AnalogSector on the robot
+ */
+public class AnalogSelectorTest {
+    @Mocked
+    AnalogInput analogInput;
+
+    @Tested
+    private AnalogSelector analogSelector;
+
+    private static double getMid(AnalogSelector.Range range) {
+        return ((range.min + range.max) / 2);
+    }
+
+    /**
+     * Tests to ensure that the autonomous selector determines the right selected index given the voltage.
+     *
+     * @param channel The channel with which the
+     * @throws NoSuchFieldException
+     * @throws IllegalAccessException
+     */
+    @Test
+    public void testGetIndex(@Injectable("0") int channel) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
It might be useful to test the minimum of one range with and the maximum of the next to ensure that they aren't the same index.